OpenRA.Mods.Common/Activities/DonateCash.cs

```diff
@@ -42,6 +42,12 @@ protected override void OnInside(Actor self)
 
 			if (self.Owner.IsAlliedWith(self.World.RenderPlayer))
 				self.World.AddFrameEndTask(w => w.Add(new FloatingText(target.CenterPosition, target.Owner.Color.RGB, FloatingText.FormatCashTick(payload), 30)));
+
+			foreach (var host in target.TraitsImplementing<INotifyCashTransfer>())
+				host.OnCashTransfer(target, self);
+
+			foreach (var me in self.TraitsImplementing<INotifyCashTransfer>())
+				me.OnCashTransfer(target, self);
```

Comment:
We end up with `Actor self` not being `self` this way, so imho this should be changed. You could for example split the `OnCashTransfer` method into two different methods: `INotifyCashTransfer.DeliveringCash` and `INotifyCashTransfer.AcceptingCash`. (Otherwise an actor having both `AcceptsDeliveredCash` and `DeliversCash` would play two sounds as well.)